features/mbedtls/targets/TARGET_NUVOTON/TARGET_M480/sha/sha1_alt.c

```diff
@@ -78,8 +71,7 @@ void mbedtls_sha1_clone(mbedtls_sha1_context *dst,
         if (src->hw_ctx.buffer_left == src->hw_ctx.blocksize) {
             mbedtls_sha1_sw_process(&dst->sw_ctx, dst->sw_ctx.buffer);
         }
-    }
-    else {
+    } else {
         // Clone S/W ctx from S/W ctx
         dst->sw_ctx = src->sw_ctx;
```

Comment:
From my current understanding, it appears that this might mix up S/W and H/W contexts, because no check is done on the type of the destination context. When this function is called, the destination context will previously have been initialized with `mbedtls_sha1_init`, and we don't know if it's H/W or S/W.

Reply:
@hanno-arm In `mbedtls_sha1_init`, `ctx` will be set to H/W ctx or S/W ctx, dependent on `crypto_sha_acquire` successful or not. It is recorded by `ctx->ishw`. In `mbedtls_sha1_clone`, if `ctx->ishw` is set to 1 (H/W ctx), a S/W ctx is cloned from H/W ctx because NUC472/M487 SHA H/W doesn't support context Save & Load.

Reply:
@ccli8 Yes, but you should ensure that the context you're cloning to wasn't set up as a hardware context - in that case, it would probably need to be destroyed and setup as a S/W context before doing the cloning.

Reply:
@hanno-arm Yes, you are right. I fix it in https://github.com/ARMmbed/mbed-os/pull/4925/commits/bfc3d27025da24274d2901daa6bdfd8a3bdbeb77.